grails-app/services/org/transmartproject/db/ontology/OntologyTermTagsResourceService.groovy

```diff
@@ -0,0 +1,33 @@
+package org.transmartproject.db.ontology
+
+import org.transmartproject.core.ontology.OntologyTerm
+import org.transmartproject.core.ontology.OntologyTermTag
+import org.transmartproject.core.ontology.OntologyTermTagsResource
+
+class OntologyTermTagsResourceService implements OntologyTermTagsResource {
+
+    @Override
+    Map<OntologyTerm, List<OntologyTermTag>> getTags(Set<OntologyTerm> ontologyTerms, boolean includeDescendantsTags) {
+        if (!ontologyTerms) {
+            return [:]
+        }
+
+        def orderedTags = I2b2Tag.createCriteria().list {
+            ontologyTerms.each { OntologyTerm term ->
+                if (includeDescendantsTags) {
+                    like 'ontologyTermFullName', (term.fullName.replaceAll('\\\\', '\\\\\\\\') + '%')
```

Comment:
This is not correct, as you're not doing anything about underscores or %. Use `asLikeLiteral'.